### ISSUE TITLE
Silverlight.pkg cannot read Info.plist.

### ISSUE THREAD

**Issue by rougegoat** [autopkg/recipes#121]
When attempting to run the Silverlight.pkg recipe(or any child recipe of it), I always receive an error that looks like this:

```
Error running recipes
File '<cache directory>/com.github.jss-recipes.jss.Silverlight/unpack/Internet Plug-Ins/Silverlight.plugin/Contents/Info.plist' does not exist or could not be read.
```

(Edit) the `<cache directory>` value does get filled in correctly, but I have opted to hide it as it has personal information in it. (/edit)

The Info.plist file both exists and is readable by all users, so this error doesn't make much sense.  The pkg is created successfully.

**Comment by BadStreff** [comment]
Looks like Microsoft changed the location of the Info plist. The new location is:
```
<string>%RECIPE_CACHE_DIR%/unpack/Library/Internet Plug-Ins/Silverlight.plugin/Contents/Info.plist</string>
```

You can create an override of the Silverlight.pkg.recipe to fix it until someone submits a pull request. I will submit one later today unless someone beats me to it :)

**Comment by BadStreff** [comment]
I created a pull request for this issue https://github.com/autopkg/recipes/pull/127

**Comment by gregneagle** [comment]
Not seeing any issue here:

```
% autopkg run -vv Silverlight.pkg
Processing Silverlight.pkg...
URLDownloader
{'Input': {'filename': u'Silverlight.dmg',
           'request_headers': {
    "user-agent" = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_3) AppleWebKit/536.28.10 (KHTML, like Gecko) Version/6.0.3 Safari/536.28.10";
},
           'url': u'http://www.microsoft.com/getsilverlight/handlers/getsilverlight.ashx'}}
URLDownloader: Storing new Last-Modified header: Thu, 19 Nov 2015 08:01:01 GMT
URLDownloader: Storing new ETag header: "801c3c6ea022d11:0"
URLDownloader: Downloaded /Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/downloads/Silverlight.dmg
{'Output': {'download_changed': True,
            'etag': '"801c3c6ea022d11:0"',
            'last_modified': 'Thu, 19 Nov 2015 08:01:01 GMT',
            'pathname': u'/Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/downloads/Silverlight.dmg',
            'url_downloader_summary_result': {'data': {'download_path': u'/Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/downloads/Silverlight.dmg'},
                                              'summary_text': 'The following new items were downloaded:'}}}
EndOfCheckPhase
{'Input': {}}
{'Output': {}}
FlatPkgUnpacker
{'Input': {'destination_path': u'/Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/expand',
           'flat_pkg_path': u'/Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/downloads/Silverlight.dmg/Silverlight.pkg'}}
FlatPkgUnpacker: Mounted disk image /Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/downloads/Silverlight.dmg
FlatPkgUnpacker: Unpacked /private/tmp/dmg.9yHtsU/Silverlight.pkg to /Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/expand
{'Output': {}}
PkgPayloadUnpacker
{'Input': {'destination_path': u'/Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/unpack',
           'pkg_payload_path': u'/Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/expand/silverlight_plugin.pkg/Payload'}}
PkgPayloadUnpacker: Unpacked /Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/expand/silverlight_plugin.pkg/Payload to /Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/unpack
{'Output': {}}
Versioner
{'Input': {'input_plist_path': u'/Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/unpack/Internet Plug-Ins/Silverlight.plugin/Contents/Info.plist',
           'plist_version_key': u'CFBundleVersion'}}
Versioner: Found version 5.1.40813.0 in file /Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/unpack/Internet Plug-Ins/Silverlight.plugin/Contents/Info.plist
{'Output': {'version': u'5.1.40813.0'}}
PkgCopier
{'Input': {'pkg_path': u'/Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/Silverlight-5.1.40813.0.pkg',
           'source_pkg': u'/Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/downloads/Silverlight.dmg/Silverlight.pkg'}}
PkgCopier: Mounted disk image /Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/downloads/Silverlight.dmg
PkgCopier: Copied /private/tmp/dmg.4fBHBQ/Silverlight.pkg to /Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/Silverlight-5.1.40813.0.pkg
{'Output': {'pkg_copier_summary_result': {'data': {'pkg_path': u'/Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/Silverlight-5.1.40813.0.pkg'},
                                          'summary_text': 'The following packages were copied:'},
            'pkg_path': u'/Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/Silverlight-5.1.40813.0.pkg'}}
Receipt written to /Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/receipts/Silverlight-receipt-20151210-103418.plist

The following packages were copied:
    Pkg Path                                                                                             
    --------                                                                                             
    /Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/Silverlight-5.1.40813.0.pkg  

The following new items were downloaded:
    Download Path                                                                                      
    -------------                                                                                      
    /Users/gneagle/Library/AutoPkg/Cache/com.github.autopkg.pkg.Silverlight/downloads/Silverlight.dmg
```

Are you running the current version of the recipe?

**Comment by BadStreff** [comment]
@gregneagle 
Yes I am running the current recipe.
Interestingly our Silverlight versions differ, I'm getting 5.1.41104.0

**Comment by gregneagle** [comment]
Oh, that's interesting. Would explain lots of weirdnesses around the Silverlight recipes...

**Comment by gregneagle** [comment]
But it would be sending that same header for you, no?

**Comment by BadStreff** [comment]
Correct, and I still get the latest version.

**Comment by gregneagle** [comment]
I think we've seen this pseudo-random behavior for a while.

Any update to the recipe needs to handle whatever "current" version MS responds with. The current recipe is broken for you, but I suspect your change would break it for me.

If someone wants to dig in and do a proper fix...

**Comment by timsutton** [comment]
I now get 5.1.41105.0, but for the past couple of months the download page has offered several different versions, and there have been open issues about this in the past.

AFAIK Silverlight registers with Microsoft Auto-Update, which means it may be able to retrieve and parse a feed to get the current version - if said version and download info is actually available in that feed (rather than simply redirecting users to the downloads page).

Since Silverlight is updated rarely and without compelling new features, seems there's less interest or need to have the recipe working more robustly. That said, there's already a fair amount of MAU-parsing code in the Office updates recipes if someone wants to try.

**Comment by BadStreff** [comment]
I suspect the same. 

My initial idea is to use the FileFinder processor, however it only returns if a file exist, not the path. So would it be better to add a custom processor to the Silverlight recipe, or add another return value to the FileFinder recipe. I could really see another return value being useful for more than just this problem.

**Comment by rougegoat** [comment]
@BadStreff I just tried out changing the input_plist_path in an override.  Looks like my change gets ignored and the base recipe's path is used instead.  What's interesting is that the error is that it says %RECIPE_CACHE_DIR% is undefined, so it uses the original which also includes %RECIPE_CACE_DIR%.  So I tried hardcoding the full file path.  It still reverted to the path in the base recipe.  I'm not sure this is something that can be fixed with an override.

**Comment by BadStreff** [comment]
@rougegoat 
Looks like you can only modify input variables with overrides, so I told you wrong. The easiest thing to do would be to create a new recipe with the correct URL. Or you can snag a copy of [mine](https://github.com/BadStreff/shsu-recipes)

**Comment by timsutton** [comment]
This should now be resolved with #134.
